ui: statements page search and pagination

vladlos opened the pull request:
Added search feature to statements page. Added paginated display (ui only).

Resolves: #41537, #41568

Release note (ui): statements page search and pagination
![Screen Shot 2019-10-16 at 5 52 09 PM](https://user-images.githubusercontent.com/12850886/66930749-b614d280-f03d-11e9-8d85-ed82c163ac4f.png)
![Screenshot 2019-11-21 at 15 08 10](https://user-images.githubusercontent.com/12850886/69340867-174d4880-0c71-11ea-8838-b98af634b651.png)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=41641) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=41641) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cockroachdb/cockroach?pullRequest=41641) it.</sub>

celiala commented on pkg/ui/src/views/app/components/Search/search.styl:
```diff
@@ -0,0 +1,61 @@
+    border none
+    font-size 14px
+  ._prefix-icon
+    color #505565
```
Update: before making variables within `.styl` file, see note below about implementing Design System palette.

celiala commented on pkg/ui/src/views/app/components/Search/search.styl:
```diff
@@ -0,0 +1,61 @@
+  ._submitted
+    .ant-input
+      &:not(:first-child)
+        padding-right 40px
```
nit: add newline

celiala commented on pkg/ui/styl/shame.styl:
```diff
@@ -252,3 +275,109 @@
+    line-height 24px
+    ._text-bold
+      color #87beff
+      font-family RobotoMono-Bold
```
nit: add newline

celiala reviewed (approved):
Nice work! LGTM, modulo some super-minor nits :)

nathanstilwell commented:
bors merge

nathanstilwell commented:
bors merge

nathanstilwell commented:
bors retry

celiala commented:
bors retry

celiala commented:
Aha -- @elkmaster , it looks like the build is failing because of this:
```
src/views/app/components/Search/index.tsx(75,11): error TS2605: JSX element type 'Input' is not a constructor function for JSX elements.
  The types returned by 'getSnapshotBeforeUpdate(...)' are incompatible between these types.
    Type 'null' is not assignable to type 'never'.
make: *** [ui-lint] Error 2
```

vilterp commented:
Why is that failing in bors and passing on GitHub CI? 🤔

vladlos committed:
ui: statements page search and pagination

Added search feature to statements page. Added paginated display (ui only).

Resolves: #41537, #41568

Release note (ui): statements page search and pagination

nathanstilwell commented:
bors merge
